
# HG changeset patch
# User Rob Lemley <rob@thunderbird.net>
# Date 1630704147 0
# Node ID 56c3c3a87360237414e838ea8fe8547d597e0ff0
# Parent  b56ce92dbc14fb3a55f771d6a85e89e73e6f25d0
Bug 1727113 - Never require that addons are signed for Thunderbird. r=john.bieling

imply_option apparently can be used to override config options that do not
have a --enable/disable flag and are set (for Firefox) only via build time
environment variables.

This will effectively disable addon signing for all Thunderbird builds
regardless of MOZ_REQUIRE_SIGNING's presence or value.

Testing on Daily is not too useful, beta or release versions should be tested
thoroughly.

Differential Revision: https://phabricator.services.mozilla.com/D124361

diff --git a/mail/config/mozconfigs/common b/mail/config/mozconfigs/common
--- a/comm/mail/config/mozconfigs/common
+++ b/comm/mail/config/mozconfigs/common
@@ -1,6 +1,3 @@
 ac_add_options --enable-application=comm/mail
 
-# Disable enforcing that add-ons are signed by the trusted root.
-MOZ_REQUIRE_SIGNING=
-
 . "$topsrcdir/build/mozconfig.common.override"
diff --git a/mail/moz.configure b/mail/moz.configure
--- a/comm/mail/moz.configure
+++ b/comm/mail/moz.configure
@@ -7,16 +7,17 @@
 set_config("MOZ_THUNDERBIRD", True)
 set_define("MOZ_THUNDERBIRD", True)
 
 imply_option("MOZ_APP_BASENAME", "Thunderbird")
 set_config("MOZ_APPUPDATE_HOST", "aus.thunderbird.net")
 imply_option("MOZ_CRASHREPORTER_URL", "https://crash-reports.thunderbird.net/")
 
 imply_option("--enable-default-browser-agent", False)
+imply_option("MOZ_REQUIRE_SIGNING", False)
 
 
 @depends(target_is_windows, target_is_linux)
 def bundled_fonts(is_windows, is_linux):
     if is_windows or is_linux:
         return True
 
 

